besspin/cyberPhys/cyberphyslib/cyberphyslib/kiosk/kiosk.py

```diff
@@ -18,141 +19,733 @@
+        """
+        self.button_pressed_next = False
+        self.button_pressed_reset = False
 
-        except Exception as exc:
-            print(f"<{self.__class__.__name__}> Error processing message: {msg}: {exc}")
+        self.hackActive(canlib.HACK_NONE)
+        self.switchActiveScenario(canlib.SCENARIO_BASELINE)
+        # Target 1 is baseline FreeRTOS
+        self.restartComponent(canlib.TARGET_1)
+        # FIXME: reset CHERI just in case
+        #self.restartComponent(canlib.TARGET_3)
+        # Infotainment server 1 is the baseline info server
+        self.restartComponent(canlib.INFOTAINMENT_SERVER_1)
+        self.restartComponent(canlib.OTA_UPDATE_SERVER_1)
+        # TODO: No need to reset Info server 2?
+        # Infotainment server 3 is in the secure ECU scenario
+        #self.restartComponent(canlib.INFOTAINMENT_SERVER_3)
+        #self.restartComponent(canlib.OTA_UPDATE_SERVER_3)
+
+        # TODO: Wait till the reset is complete?
+        # Reset state
+        self.hack12_has_been_initialized = False
+        #  Respond
+        self.ipc_msg['status'] = 200 # OK
+
+    @page
+    def hack02_kiosk_intro_enter(self, arg):
+        """
+        Reset is complete
+        No action needed
+        """
+        self.button_pressed_next = False
+        # Respond
+        self.ipc_msg['status'] = 200 # OK
+
+    @page
+    def hack05_info_attempt_enter(self, arg):
+        """
+        Hack OTA server
+        * hack the server
+        * upload hacked infotainment
+        """
+        self.button_pressed_next = False
+        hack_ok = self.hack_ota_and_upload_hacked_infotainment_server()
+        self.ipc_msg['retval'] = hack_ok
+        self.ipc_msg['status'] = 200 # OK
+
+    @page
+    def hack06_info_exploit_enter(self, arg):
+        """
+        Wait for the exploit selection
+        """
+        self.button_pressed_next = False
+        self.exploit_complete = False
+
+        self.ipc_msg['status'] = 200 # OK
+
+    @page
+    def hack06_info_exploit_attemp_hack_enter(self, arg):
+        """
+        Attemp a selected infotainment exploit
+        * check args for which hack to use
+        """
+        self.button_pressed_info_exploit = False
+        self.exploit_complete = True
+        # `ipc_msg` is updated accordingly
+        self.execute_infotainment_hack(arg)
+
+        self.ipc_msg['status'] = 200 # OK
+
+
+    @page
+    def hack08_critical_exploit_enter(self, arg):
+        """
+        Attemp a selected critical exploit
+        * check args for which hack to use
+        """
+        self.button_pressed_critical_exploit = False
+        self.exploit_complete = True
+
+        # `ipc_msg` is updated accordingly
+        self.execute_ecu_hack(arg)
+
+        self.ipc_msg['status'] = 200 # OK
+
+    @page
+    def hack09_protect_enter(self, arg):
+        """
+        Switch to SSITH_INFOTAINMENT_SCENARIO
+        NOTE: Reset baseline target(s) here to save time?
```

Comment:
Yes, I think that's a good idea, but can be implemented later on during fine tuning.

Reply:
I will keep the note there.